hnn_core/tests/test_gui.py

```diff
@@ -734,20 +752,33 @@ def test_gui_add_tonic_input(setup_gui):
     gui.add_drive_button.click()
 
     # Check last drive (Tonic)
-    last_drive_pos = len(gui.drive_widgets) - 1
-    assert gui.drive_widgets[last_drive_pos]['type'] == "Tonic"
-
-    gui.drive_widgets[last_drive_pos]['amplitude']["L5_pyramidal"].value = 10
-    gui.drive_widgets[last_drive_pos]['t0'].value = 0
-    gui.drive_widgets[last_drive_pos]['tstop'].value = 15
+    last_drive = gui.drive_widgets[-1]
+    assert last_drive['type'] == "Tonic"
+    assert last_drive['t0'].value == 0.0
+    assert last_drive['tstop'].value == 170.0
+    assert last_drive['amplitude']['L5_pyramidal'].value == 0
+
+    # Set new widget values
+    last_drive['t0'].value = 0
+    last_drive['tstop'].value = 15
+    last_drive['amplitude']['L5_pyramidal'].value = 10
+
+    # Check that you can't add more than one tonic
+    gui.add_drive_button.click()
```

Comment:
Is the current behavior that the previous tonic drive is overwritten?

Reply:
Current behavior is that you cannot add more than one tonic. If you click the button again nothing will happen. It won't overwrite the previous tonic widget. There should probably be a console message though...